core/src/main/scala/geotrellis/store/query/RepositoryM.scala

```diff
@@ -16,21 +16,38 @@
 
 package geotrellis.store.query
 
+import cats.{~>, Applicative, Id, Monoid, MonoidK, SemigroupK}
 import cats.syntax.apply._
 import cats.syntax.semigroupk._
-import cats.{FlatMap, Semigroup, SemigroupK}
 
-trait RepositoryM[M[_], G[_], T] {
+trait RepositoryM[M[_], G[_], T] { self =>
   def store: M[G[T]]
   def find(query: Query): M[G[T]]
+
+  def mapK[F[_]](f: M ~> F): RepositoryM[F, G, T] =
+    new RepositoryM[F, G, T] {
+      def store: F[G[T]]              = f(self.store)
+      def find(query: Query): F[G[T]] = f(self.find(query))
+    }
 }
 
 object RepositoryM {
-  implicit def semigroupRepositoryM[M[_]: FlatMap, G[_]: SemigroupK, T]: Semigroup[RepositoryM[M, G, T]] =
-    Semigroup.instance { (l, r) =>
-      new RepositoryM[M, G, T] {
-        def store: M[G[T]] = l.store.map2(r.store)(_ <+> _)
-        def find(query: Query): M[G[T]] = l.find(query).map2(r.find(query))(_ <+> _)
-      }
+  def empty[M[_]: Applicative, G[_]: MonoidK, T]: RepositoryM[M, G, T] =
+    new RepositoryM[M, G, T] {
+      def store: M[G[T]]              = Applicative[M].pure(MonoidK[G].empty)
+      def find(query: Query): M[G[T]] = store
     }
+
+  implicit def monoidRepositoryM[M[_]: Applicative, G[_]: MonoidK: SemigroupK, T]: Monoid[RepositoryM[M, G, T]] =
+    new Monoid[RepositoryM[M, G, T]] {
+      def empty: RepositoryM[M, G, T]                                                     = RepositoryM.empty[M, G, T]
+      def combine(l: RepositoryM[M, G, T], r: RepositoryM[M, G, T]): RepositoryM[M, G, T] =
+        new RepositoryM[M, G, T] {
+          def store: M[G[T]]              = l.store.map2(r.store)(_ <+> _)
+          def find(query: Query): M[G[T]] = l.find(query).map2(r.find(query))(_ <+> _)
+        }
+    }
+
+  implicit def idToApplicative[M[_]: Applicative, G[_], T](repository: RepositoryM[Id, G, T]): RepositoryM[M, G, T] =
```

Comment:
Given the implicit resolution woes from yesterday, I'm noticing that this is a very generic name. Perhaps change to something like `repositoryMIdToApplicative`?